relay-server/src/actors/upstream.rs

```diff
@@ -458,6 +483,53 @@ impl UpstreamRelay {
             .spawn(ctx);
     }
 
+    /// Adds a metric for the upstream request.
+    fn meter_result(
+        request: &UpstreamRequest,
+        send_result: &Result<ClientResponse, UpstreamRequestError>,
+    ) {
+        let status_code: StatusCode;
+        let mut sc = "-";
+        let result: &str;
+        match send_result {
+            Ok(ref client_response) => {
+                status_code = client_response.status();
+                sc = status_code.as_str();
+                result = "success";
+            }
+            Err(UpstreamRequestError::ResponseError(status_code, _)) => {
+                sc = status_code.as_str();
+                result = "response_error";
+            }
+            Err(UpstreamRequestError::PayloadFailed(_)) => result = "payload_failed",
+            Err(UpstreamRequestError::SendFailed(_)) => result = "send_failed",
+            Err(UpstreamRequestError::RateLimited(_)) => result = "rate_limited",
+            Err(UpstreamRequestError::NoCredentials)
+            | Err(UpstreamRequestError::ChannelClosed)
+            | Err(UpstreamRequestError::BuildFailed(_))
+            | Err(UpstreamRequestError::InvalidJson(_)) => {
+                // these are not errors caused when sending to upstream so we don't need to log anything
+                log::error!("meter_result called for unsupported error");
+                return;
+            }
+        }
+
+        let retries = match request.previous_retries {
+            0 => "0",
+            1 => "1",
+            x if x < 5 => "few",
+            _ => "many",
+        };
```

Comment:
@beezz What's the best way to plot such tags on DataDog? I'm wondering if we could rather have this as histogram with tags. Note that the histogram comes with an implicit counter, so we get both in one.

Reply:
So what would the histogram measure... number of tries? And then look at the different percentiles to see on average how many requests are retried. I think/hope that we should rarely have retries so just showing them as counters and being able to see when we have more than one retries is enough (but that is just my gut feeling).

Reply:
A proper histogram could help us identify conditions where we start spinning in retries. In addition to the histogram, you *also* get a plain counter (essentially every time you log the histogram it registers one count).

Reply:
Let's have a quick chat about this.

Reply:
@jan-auer I'd probably plot it as stacked bar/area in DD to properly see the distribution. Can you elaborate on the histogram idea? My concern there is that it will disjoin the retries tag/counter from the rest of the tags (route, resp. code etc.).